network/src/transport.rs

```diff
@@ -120,6 +120,17 @@ impl ConnectionMetadata {
             application_protocols,
         }
     }
+
+    pub fn mock(peer_id: PeerId) -> ConnectionMetadata {
```

Comment:
should be able to `#[cfg(any(test, feature = "testing", feature = "fuzzing"))]` here